os/pm/pm_idle.c

```diff
@@ -65,22 +65,62 @@ void pm_idle(void)
 	enum pm_state_e newstate;
 #ifdef CONFIG_PM_TIMEDWAKEUP
 	clock_t delay;
+#endif
+#ifdef CONFIG_SMP
+	int me = this_cpu();
+	int cpu;
+	int paused_count = 0;
+	FAR struct tcb_s *tcb = NULL;
 #endif
 	/* Decide, which power saving level can be obtained */
 	flags = enter_critical_section();
 	newstate = pm_checkstate();
 	/* Perform state-dependent logic here */
-	pmvdbg("newstate= %d\n", newstate);
+	/* For SMP case, we need to check secondary core status
+	   If secondary core status is not in idle thread, abort
+	   the sleep and check again on next cycle
+	*/
+#ifdef CONFIG_SMP
+	if (newstate == PM_SLEEP) {
+		for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
+			/* If the CPU is just back from sleep, abort the sleep */
+			if (up_get_secondary_cpu_state(cpu) == CPU_WAKE_FROM_SLEEP) {
+				goto EXIT;
+			}
+			tcb = current_task(cpu);
+			/* Lock the cpu first, before checking which task it is handling */
+			(void)up_cpu_pause(tcb);
```

Comment:
i think, we need to bring tcb after pause so that can be protected from context switching.
```suggestion
			/* Lock the cpu first, before checking which task it is handling */
			(void)up_cpu_pause(cpu);
			tcb = current_task(cpu);

```